src/main/java/codeu/controller/RegisterServlet.java

```diff
@@ -12,10 +17,46 @@
 
 public class RegisterServlet extends HttpServlet
 {
+
+  private UserStore userStore;
+
+  @Override
+  public void init() throws ServletException {
+    super.init();
+    setUserStore(UserStore.getInstance());
+  }
+
+  void setUserStore(UserStore userStore) {
+    this.userStore = userStore;
+  }
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response)
-  throws IOException, ServletException
-  {
+  throws IOException, ServletException {
     request.getRequestDispatcher("/WEB-INF/view/register.jsp").forward(request, response);
   }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response)
+    throws IOException, ServletException {
+
+    String username = request.getParameter("username");
+    String password = request.getParameter("password");
+
+    if (!username.matches("[\\w*\\s*]*")) {
```

Comment:
One thing to think about: the default restriction was that usernames can only contain letters and numbers. Do you want to keep this restriction? This is completely up to the team and how you want your chat app to behave.

(You don't have to do anything now, just something to think about and talk about with the team.)

Reply:
I'll keep this in mind!